pkg/virt-controller/watch/vmi_test.go

```diff
@@ -3021,6 +3024,100 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			Expect(controller.cidsMap.reverse).To(BeEmpty())
 		})
 	})
+
+	Context("dynamic interface attachment", func() {
```

Comment:
Can you please avoid adding this to this huge file? Just opening a new test file for the hotplug will be nice.

Reply:
Another point is, that at the level of VMI controller reconcile loop, I would just be interested to see the hotplug is examined (processed), not to see any details of what it does exactly. If we follow this, it allows to adjust the high level abstractions without entering details. (e.g. when it is called, how many times, etc)

The logic under the hood, can be tested in the lower level handlers.

But maybe the problem is that everything is done on one place, with no abstraction, causing additions and continue the trend with nothing being able to stop it.

Reply:
Just want to make it clear the minimum is obviously leaving this here. 

Given the current PR is opened for 15 months, has 554 comments in it, I wholeheartedly suggest this should be moved in a follow-up.